stories/link.fmf

```diff
@@ -0,0 +1,49 @@
+/test:
+    story:
+        As a tester I want to link test verifying Jira issue.
+
+    description:
+        For cases when there is a single test verifying given issue there would be fmf
+        identifier pointing to that individual test.
+
+    example:
+      - |
+        url-options: ?test_url=https://github.com/teemtee/tmt&test-name=/tests/core/smoke
+
+    link:
+      - https://issues.redhat.com/browse/TT-207
+
+/plan:
+    story:
+        As a tester I want to link plan verifying Jira issue.
+
+    description:
+        For cases when there is a single plan verifying given issue there would be fmf
+        identifier pointing to that individual plan, where all the test executed under the given plan are
+        covering the issue.
+
+    example:
+      - |
+        url-options: ?plan-url=https://github.com/teemtee/tmt&plan-name=/plans/features/basic&type=plan
+
+    link:
+      - https://issues.redhat.com/browse/TT-207
+
+/test-and-plan:
+    story:
+        As a tester I want to link test and plan verifying Jira issue.
+
+    description:
+        If the issue is covered by a tmt test which should be executed
+        under a specific tmt plan, fmf identifiers of both test and
+        plan would be provided to select what exactly would be
+        executed. This should allow faster test execution as the whole
+        tmt plan does not have to be executed.
+
+    example:
+      - |
+        url-options: ?test-url=https://github.com/teemtee/tmt&test-name=/tests/core/smoke&?plan-url=https://github.com/teemtee/tmt&plan-name=/plans/features/basic&type=plan
```

Comment:
```suggestion
    https://tmt.org/?test-name=/tests/core/smoke&plan-url=https://github.com/teemtee/tmt&plan-name=/plans/features/basic
```

Test url could be omitted as most often we just pick the test by name. Perhaps we could include both example with and without `test-url`.

Reply:
Hm, do we even need to allow `test-url` with `plan-url` specified? The test has to be discoverable by the plan, there is no support to inject test-url into the plan, or is it?

Reply:
What about the case when `url` is specified in the `discover` step? Which means, you have a plan in `repo-one` and test in `repo-two` and ideally you would like to see both test and plan details on one click from Jira. On the server, both `repo-one` and `repo-two` should be cloned and information extracted from them.

Reply:
Maybe just stating obvious but this design assumes that individual URLs passed as parameters won't themselves contain parameters. If this is intentional limitation it should be documented.

Reply:
Special characters could be escaped, so I guess, even the git repo url could include parameters if they were needed. Or did you mean anything else?

Reply:
But how would you know that git repo URL should be un-escaped?

Reply:
It **always** needs to be "un-escaped". The `plan-url` field will **never** look like the one in the example, `plan-url=https://github.com/teemtee/tmt`, *when used in the actual URL* - the actual one would look like this: `plan-url=https%3A//github.com/teemtee/tmt`. If the plan URL needs to contain any parameters, e.g. `https://github.com/teemtee/tmt?ref=main`, the eventual `plan-url` value shall again be converted for URL use, using `urllib.parse.quote()`-like function, and we would get this: `plan-url=https%3A//github.com/teemtee/tmt%3Fref%3Dmain`.

Whether to un-escape the value or not is not in question: it should happen always because values must be encoded properly before they can be used in URL. Every sane HTTP library like `requests` will take care of it.

Reply:
Ack

Reply:
Let's update the example link accordingly, to prevent confusion here.